nemo/collections/vlm/qwen2vl/model/base.py

```diff
@@ -0,0 +1,1096 @@
+        self.add_encoder = add_encoder
+        self.add_decoder = add_decoder
+
+        self.encoder_hidden_state = None
+        self.vision_model = None
+        self.vision_projection = None
+        self.language_model = None
+
+        self.sequence_parallel_lm = language_transformer_config.sequence_parallel
+        self.tp_comm_overlap_lm = language_transformer_config.tp_comm_overlap
+
+        self.share_embeddings_and_output_weights = False
+
+        if self.add_decoder:
+            self.language_model = language_transformer_config.configure_model(
+                tokenizer=tokenizer,
+                pre_process=pre_process,
+                post_process=post_process,
+            )
+
+            self.share_embeddings_and_output_weights = self.language_model.share_embeddings_and_output_weights
+            self._language_max_sequence_length = self.language_model.max_sequence_length
+            self._language_is_pipeline_parallel = language_transformer_config.pipeline_model_parallel_size > 1
+            if config.language_model_from_pretrained is not None:
+                sharded_state_dict = dict(state_dict=self.language_model.sharded_state_dict(prefix="module."))
+                loaded_state_dict = dist_checkpointing.load(
+                    sharded_state_dict=sharded_state_dict,
+                    checkpoint_dir=ckpt_to_weights_subdir(config.language_model_from_pretrained, is_saving=False),
+                    validate_access_integrity=False,
+                )
+                loaded_state_dict = {k.removeprefix("module."): v for k, v in loaded_state_dict["state_dict"].items()}
+                self.language_model.load_state_dict(loaded_state_dict)
+                logging.info(f"Restored language model weights from {config.language_model_from_pretrained}")
+        else:
+            if config.language_model_from_pretrained is not None:
+                dist_checkpointing.load(
+                    sharded_state_dict=dict(state_dict={}),
+                    checkpoint_dir=config.language_model_from_pretrained,
+                    validate_access_integrity=False,
+                )
+
+        if self.add_encoder:
+            self.vision_model = vision_transformer_config.configure_model()
+            self.vision_projection = vision_projection_config.configure_model()
+            self._drop_vision_class_token = drop_vision_class_token
+
+        self.freeze(
+            freeze_language_model=config.freeze_language_model,
+            freeze_vision_model=config.freeze_vision_model,
+            freeze_vision_projection=config.freeze_vision_projection,
+        )
+
+        self.model_type = ModelType.encoder_or_decoder
+        # This attribute is needed to check if an all-reduce is required
+        # on the word embeddings inside `finalize_model_grads._allreduce_word_embedding_grads`.
+
+        self.vision_model_from_hf = str(self.vision_model.__class__.__module__).startswith("transformers.")
```

Comment:
do you want to support different HF vit as vision encoder?
If this is not really get used, just remove it.